Extending/Packages/Packages-on-Umbraco-Cloud/index.md

```diff
@@ -0,0 +1,287 @@
+
+:::warning
+
+In this example there would be no way for Deploy to know to also transfer the image. We assume that you would transfer all content and images to ensure it is on the target environment under a different ID.
+
+That is not a good assumption, and you may have noticed that there is a parameter on the `ToArtifact` method that you could update by finding the image and adding it to `ICollection<ArtifactDependency> dependencies`. 
+:::
+
+In order to convert to a GUID in the `ToArtifact` method, we will update the code:
+
+```csharp
+public string ToArtifact(object value, PropertyType propertyType, ICollection<ArtifactDependency> dependencies)
+{
+    var svalue = value as string;
+    if (string.IsNullOrWhiteSpace(svalue))
+        return null;
+
+    if (int.TryParse(svalue, out var intvalue))
+        return null;
+
+    var getKeyAttempt = _entityService.GetKey(intvalue, UmbracoObjectTypes.Media);
+
+    if (getKeyAttempt.Success)
+    {
+        var udi = new GuidUdi(Constants.UdiEntityType.Media, getKeyAttempt.Result);
+        dependencies.Add(new ArtifactDependency(udi, false, ArtifactDependencyMode.Exist));
+
+        return udi.ToString();
+    }
+    else
+    {
+        _logger.Debug<BadMediaValueConnector>($"Couldn't convert integer value #{intvalue} to UDI");
+    }
+
+    return string.Empty;
+}
+```
+
+You can find references on the methods used here in our API documentation:
+
+<!-- vale off -->
+
+- [EntityService.GetKey](https://our.umbraco.com/apidocs/v8/csharp/api/Umbraco.Core.Services.Implement.EntityService.html#Umbraco_Core_Services_Implement_EntityService_GetKey_System_Int32_Umbraco_Core_Models_UmbracoObjectTypes_)
+- [new GuidUdi](https://our.umbraco.com/apidocs/v8/csharp/api/Umbraco.Core.GuidUdi.html#Umbraco_Core_GuidUdi__ctor_System_String_System_Guid_)
+- [new ArtifactDependency](https://our.umbraco.com/apidocs/v8/csharp/api/Umbraco.Core.Deploy.ArtifactDependency.html#Umbraco_Core_Deploy_ArtifactDependency__ctor_Umbraco_Core_Udi_System_Boolean_Umbraco_Core_Deploy_ArtifactDependencyMode_)
+
+<!-- vale on -->
+
+When stepping through the code we can see that everything seems to work fine:
+
+![Stepped through code](images/steppingThroughCode.png)
+
+:::note
+Note: Showing the variable values is a feature of [ReSharper](https://www.jetbrains.com/resharper/) .
+:::
+
+By the time we hit `FromArtifact` value of `"umb://media/00c9eff861654f52be7a33367c3561a4"` all that is left to do is convert back to an `int`.
+
+```csharp
+public object FromArtifact(string value, PropertyType propertyType, object currentValue)
+{
+    if (string.IsNullOrWhiteSpace(value))
+        return value;
+
+    if (!GuidUdi.TryParse(value, out var udi) || udi.Guid == Guid.Empty)
+        return value;
+
+    var getIdAttempt = _entityService.GetId(udi.Guid, UmbracoObjectTypes.Media);
+
+    if (!getIdAttempt.Success) return value;
```

Comment:
It isn't really correct to return the value if the media item couldn't be resolved. This will result in a GUID going into a value of a property editor expecting a integer ID - which will in many cases cause this property editor to break or maybe even break the backoffice in a way where you can't edit the item.

It would be better to return a empty value / null or whatever the property editor has when it doesn't have a value.

Whatever is returned from this method is what goes into the value of the property editor .... keep that in mind :)

Reply:
Good point, changed these two checks to return null:

```
if (!GuidUdi.TryParse(value, out var udi) || udi.Guid == Guid.Empty)
                return null;

if (!getIdAttempt.Success) return null;
```